lib/everypolitician/popolo/collection.rb

```diff
@@ -35,11 +37,21 @@ def where(attributes = {})
         attributes.map { |k, v| index_for(k.to_sym)[v].to_a }.reduce(:&) || []
       end
 
+      def of_class(klass)
+        @of_class[klass] ||= select { |e| e.class == klass }
+      end
+
       private
 
       def index_for(attr)
         @indexes[attr] ||= group_by(&attr)
       end
+
+      def class_for_entity(document)
+        @event_class[document[:classification]] ||= self.class.entity_class.subclasses.select do |e|
```

Comment:
I think this method should just return `self.class.entity_class` in the `Collection` class, then it can be overridden in the `Events` class to check for `document[:classification]`, which is an event-specific field.